cmd/clusterctl/client/repository/metadata_client_test.go

```diff
@@ -139,16 +142,13 @@ func Test_metadataClient_Get(t *testing.T) {
 				repository: tt.fields.repository,
 			}
 			got, err := f.Get()
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("got = %v, want %v", got, tt.want)
-			}
+			g.Expect(err).NotTo(HaveOccurred())
```

Comment:
nit we should be consistent in having assertion on the operation err/on wantErr like in Test_componentsClient_Get